src/gov/nist/javax/sip/stack/SIPClientTransactionImpl.java

```diff
@@ -1545,7 +1544,7 @@ public void processResponse(SIPResponse sipResponse, MessageChannel incomingChan
     int code = sipResponse.getStatusCode();
     boolean isRetransmission = !responsesReceived.add(Integer.valueOf(code));
     if (code > 100 && code < 200 && isRetransmission) {
-      if (lastResponse != null && !sipResponse.toString().equals(lastResponse.toString())) {
+      if (lastResponse != null && !sipResponse.equals(lastResponse)) {
```

Comment:
As you noticed - equals() (in addtion to toStrong().equals() ) also compars the headers. Thus when the message content (ex. SDP will change) then the Content-Length header shall also change. I thik this shall be enough for the majority of cases. The general assumption is that if any header of the SIP response changes - it means that this is not a retransmitted response as allowed by SIP spec.